web3c/index.js

```diff
@@ -16,6 +18,8 @@ module.exports = function (provider) {
   localWeb3.call(this, provider);
   if (this.version && !this.version.api) { // v1.0 series
     this.confidential = new Confidential(this, localStorage, MraeBox);
+    this.oasis = new Oasis(this);
```

Comment:
can we have web3.oasis namespace be the non-confidential but with expiry / other oasis-specific functionality namespace?
(i'm wondering if there's a way to avoid monkey-patching the web3.eth namespace entirely)

Reply:
So we would put the contract object on the oasis namespace if we did this.

Reply:
From an api perspective, I would prefer not doing this. It’s less clear to have three options for contract objects. The eth/confidential dichotomy is obvious for non confidential and confidential.

Adding a third into the mix is not clearly a good idea to me.

Reply:
A more dramatic change would be to drop the eth namespace and just use Oasis. We would still need to do the copying though.

Reply:
that's what i was suggesting - eth would be the unpatched web3 eth. Oasis would be the patched eth + our options. confidential would be oasis but defaulting to confidential header tagged on.

Reply:
I dig it.

Reply:
Though we then lose the web3 mapping of function to rpc method. E.g., web3.oasis.getBlock => eth_getBlock.

Reply:
we've already lost that with confidential, though 🤔

Reply:
We don't. Only things on confidential are `[ 'keyManager', 'getPublicKey', 'call', 'Contract', 'accounts' ]`.

The only rpcs there are `getPublicKey` and `call`. 

The `Contract` object doesn't map to any rpcs. 1-1 (not even in eth namespace) since it's an object abstraction, not rpc.